src/core/Form/Checkbox/Checkbox.tsx

```diff
@@ -66,50 +118,136 @@ class DefaultCheckbox extends Component<CheckboxProps> {
 
   render() {
     const {
-      children,
+      id,
+      className,
       disabled = false,
-      onClick,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      'aria-describedby': ariaDescribedBy,
+      children,
+      checked: dismissChecked,
+      defaultChecked: dismissDefaultChecked,
+      onClick: dismissOnClick,
+      hintText,
       status,
+      statusText,
+      name,
+      value,
+      onClick,
       variant,
-      className,
       ...passProps
-    } = withSuomifiDefaultProps(this.props);
+    } = this.props;
+
     const { checkedState } = this.state;
+
+    if (!children) {
+      logger.error(
+        'Checkbox component should have a label or a child element that acts as one. Add label content or a child element.',
+      );
+    }
+
+    if (
+      ('name' in this.props && (typeof name !== 'string' || name === '')) ||
+      ('value' in this.props && (typeof value !== 'string' || value === ''))
+    ) {
+      logger.warn(
+        'Name and value props should have non-empty values if provided.',
+      );
+    }
+
+    const statusTextId = `${id}-statusText`;
+    const hintTextId = `${id}-hintText`;
+
+    const getDescribedBy = () => {
+      if (statusText || hintText || ariaDescribedBy) {
+        return {
+          'aria-describedby': [
+            ...(statusText ? [statusTextId] : []),
+            ...(hintText ? [hintTextId] : []),
+            ...(ariaDescribedBy ? [ariaDescribedBy] : []),
+          ].join(' '),
+        };
+      }
+      return {};
+    };
+
+    const newCheckboxInputProps = {
```

Comment:
Id prefer to call these derived or extendedCheckboxInputProps rather than new. Would indicate the use case better.

Reply:
Or should we get rid of this and told it in directly where the component is?

```
<HtmlInput type="checkbox" 
  disabled,
  id,
  'aria-label': ariaLabel,
  'aria-labelledby': ariaLabelledBy,
  'aria-invalid': status === 'error',
  checked: !!checkedState,
  className: checkboxClassNames.input,
  onChange: this.handleClick,
  name,
  ...getDescribedBy(),
  ...(value ? { value } : {}),
/>
```

As these props are only for input and not used for anything else.

Reply:
Makes sense! Leaving the complex getDescribedBy etc. logic outside of the template is good, but separating the props completely from the context is unnecessary.

Reply:
This is now changed.